### DISCUSSION TITLE
Resampled Efficient Frontier

### DISCUSSION THREAD

**Discussion by Pierre-22** [dcajasn/Riskfolio-Lib#205] [Q&A]
Hi,
Does anyone know how to apply a resampled efficient frontier using this library. I looked through the examples on  the site but couldn’t find anything. Thanks!

**Comment by dcajasn** [comment]
Hi @Pierre-22,

It is not implemented because in my experience it does not provide any benefit (it doesn't increase diversification or increase robustness significantly) and only increase computational cost. You can implement it by calculating the efficient frontier for several synthetic returns and averaging the weights for each point in the frontier. 

Best regards,
Dany

**Reply by antonvorobets** [reply]
This answer is clearly incorrect. The increased diversification is obvious from resampled optimization. Hence, it is the method most frequently used by asset allocation investors in practice. But it does require that the portfolio optimizers are fast and stable of course :-)

For more on resampled optimization, see: https://ssrn.com/abstract=4709317

**Reply by dcajasn** [reply]
Hi @antonvorobets,

You are wrong, for the following reasons:

- Robustness: refers to the stability of the results of an optimization problem under slight changes in the input parameters. The input parameters of the resampled portfolio are the mean vector and covariance matrix. Following Michaud, these parameters are used to generate random samples from a normal distribution that are used as inputs of several optimization processes that are averaged to calculate the resampled portfolio. However, if you add small variations to the initial parameters, the resulting random samples will generate a different resampled portfolio that could be very different than the original resampled portfolio. So, in the end, you are not solving the robustness problem of the mean variance portfolio. You can check that doing a simple simulation exercise comparing the box plot of the optimal weights of the mean variance portfolio and the resampled mean variance portfolio for several variations of initial mean vector and covariance matrix.

- Diversification: refers to the number of assets that make up an investment portfolio. The diversification of a portfolio increases as the number of assets increases. The increase of diversification is too small compared to other methods used to increase diversification. If you calculate the number of effective assets, they are practically the same. 

Best,
Dany

**Reply by antonvorobets** [reply]
Two comments:

1. Portfolio optimization is much more than mean-variance, which I assume you can agree to given the number of risk measures in this package and your introduction of new ones. In fact, sophisticated institutional asset allocators focus on minimizing tail risks (CVaR).

2. I will let logically thinking people decide what they think is correct between the article I shared and your comments :-)

**Reply by dcajasn** [reply]
CVaR doesn't account for tail risk, It tries but in the end is not designed to do that because is a weighted average of losses beyond the VaR, in the end CVaR give more weight to losses that are near to VaR because they have higher probability. 

The risk measures that account for tail risk are the Entropic Value at Risk, Relativistic Value at Risk, Maximun Loss and Kurtosis. You can check my presentation at CEQURA conference where I explain that ideas here https://github.com/dcajasn/CEQURA-Conference-2023/blob/main/CEQURA%20Slides%20-%20RLVaR%20.pdf

**Reply by antonvorobets** [reply]
Agree to disagree :-)

**Comment by Pierre-22** [comment]
Hi Dany,
Appreciate your response, I will do what you suggested. Great library by
the way, very useful!
